app/_src/kubernetes-ingress-controller/guides/high-availability/fallback-config.md

```diff
@@ -0,0 +1,327 @@
+(that will enable us to slip a broken configuration to the controller):
+
+```bash
+helm upgrade --install kong kong/ingress -n kong \
+  --set ingressController.env.feature_gates=FallbackConfiguration=true \
+  --set ingressController.admissionWebhook.enabled=false
+```
+
+In the example, we'll consider a situation where:
+
+1. We have two `HTTPRoutes` pointing to the same `Service` and working correctly.
+2. One of the `HTTPRoutes` gets broken because of referencing an invalid `KongPlugin`.
+
+First, let's deploy the `Service` and its backing `Deployment`:
+
+```bash
+kubectl apply -f {{site.links.web}}/assets/kubernetes-ingress-controller/examples/echo-service.yaml
+```
+The results should look like this:
+```text
+service/echo created
+deployment.apps/echo created
+```
+
+Next, let's deploy the `HTTPRoutes`:
+
+```bash
+echo 'apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: route-a
+spec:
+  parentRefs:
+  - name: kong
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /route-a
+    backendRefs:
+    - name: echo
+      port: 1027
+---
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: route-b
+spec:
+  parentRefs:
+  - name: kong
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /route-b
+    backendRefs:
+    - name: echo
+      port: 1027
+      ' | kubectl apply -f -
+```
+
+The results should look like this:
+```text
+httproute.gateway.networking.k8s.io/route-a created
+httproute.gateway.networking.k8s.io/route-b created
+```
+
+Let's ensure that the `HTTPRoutes` are working as expected:
+
+```bash
+curl -i $PROXY_IP/route-a
+```
+
+The results should look like this:
+```text
+HTTP/1.1 200 OK
+Content-Type: text/plain; charset=utf-8
+Content-Length: 137
+Connection: keep-alive
+X-Kong-Upstream-Latency: 1
+X-Kong-Proxy-Latency: 1
+Via: kong/3.6.0
+X-Kong-Request-Id: 5bf50016730eae43c359c17b41dc8614
+
+Welcome, you are connected to node orbstack.
+Running on Pod echo-74c66b778-szf8f.
+In namespace default.
+With IP address 192.168.194.13.
+```
+
+```bash
+curl -i $PROXY_IP/route-b
+```
+
+The results should look like this:
+```text
+HTTP/1.1 200 OK
+Content-Type: text/plain; charset=utf-8
+Content-Length: 137
+Connection: keep-alive
+X-Kong-Upstream-Latency: 0
+X-Kong-Proxy-Latency: 0
+Via: kong/3.6.0
+X-Kong-Request-Id: 14ae28589baff9459d5bb3476be6f570
+
+Welcome, you are connected to node orbstack.
+Running on Pod echo-74c66b778-szf8f.
+In namespace default.
+With IP address 192.168.194.13.
+```
+
+Now, let's introduce a breaking change in the `route-2` by attaching a broken `KongPlugin` to it.
+
+Create a `KongPlugin` with an invalid configuration:
+
+```bash
+echo 'apiVersion: configuration.konghq.com/v1
+kind: KongPlugin
+metadata:
+  name: key-auth
+plugin: key-auth
+config:
+  # Should be 'key_names', not 'keys'.
+  keys: ["key"]' | kubectl apply -f -
+```
```

Comment:
This plugin fails admission webhook validation:

```
Error from server: error when creating kubernetes-ingress-controller/examples/gateway-httproute.yaml": admission webhook "validations.kong.konghq.com" denied the request: plugin failed schema validation: schema violation (config.keys: unknown field)
```

Don't we want to provide an example that will pass that stage and fail during Gateway configuration apply phase (as noted in the `How it works` section above, point 3)?

Reply:
Yeah, that's why I added the step to disable the webhook with an explanation saying it will allow us to slip invalid configuration to stage 3. As we discussed once, I think it will be more _reliable_ to demonstrate it this way (turning the webhook off and pushing something that the webhook would reject). If we knew there was a problem that we could trigger with the webhook turned on, we should rather plumb it in the webhook or the translation phase, not make it an example in the guide, right? 😄

Reply:
👍 fair point.

Reply:
I came here to say this. We should provide an example that can actually happen (such as duplicate plugins on a single service) rather than disabling the admission webhook

Reply:
@mheap My rationale for doing it this way was that by disabling the webhook server we can reliably showcase this feature. If we have any known issue, we tend to fix that either in the webhook or the translation as soon as possible, therefore using a known issue as an example doesn't make much sense. In fact, duplicate plugins on a single service have already been covered in the webhook:
```
Error from server: admission webhook "validations.kong.konghq.com" denied the request: Service has invalid KongPlugin annotation: cannot attach multiple plugins: key-auth-1, key-auth-2 of the same type key-auth
```

Reply:
Just for broader context, it doesn't work every time see
- https://github.com/Kong/kubernetes-ingress-controller/issues/5995

and this is connected with the behavior described in
- https://github.com/Kong/kubernetes-ingress-controller/issues/6139 

so I think that in the future we'll cover it with webhook too

Reply:
I get an event, but it isn't blocked by the admission controller for me:

```
default       0s          Warning   KongConfigurationApplyFailed     kongplugin/rate-limiting-authenticated         invalid plugin:rate-limiting: uniqueness violation: 'plugins' entity with primary key set to '35619164-260d-5ac2-a840-b86e6cc87f76' already declared
```

```
❯ kaf .
service/echo created
deployment.apps/echo created
ingress.networking.k8s.io/echo created
secret/alex-key-auth created
kongconsumer.configuration.konghq.com/alex created
secret/bob-key-auth created
kongconsumer.configuration.konghq.com/bob created
kongplugin.configuration.konghq.com/key-auth-example created
kongplugin.configuration.konghq.com/rate-limiting-base created
kongplugin.configuration.konghq.com/rate-limiting-authenticated created
```

Notice the commented out annotation on the `alex` consumer, which makes the rate limit plugin on the `echo` service a duplicate:

```
❯ kaf .
service/echo created
deployment.apps/echo created
ingress.networking.k8s.io/echo created
secret/alex-key-auth created
kongconsumer.configuration.konghq.com/alex created
secret/bob-key-auth created
kongconsumer.configuration.konghq.com/bob created
kongplugin.configuration.konghq.com/key-auth-example created
kongplugin.configuration.konghq.com/rate-limiting-base created
kongplugin.configuration.konghq.com/rate-limiting-authenticated created
```

```
apiVersion: v1
kind: Service
metadata:
  labels:
    app: echo
  name: echo
  annotations:
    konghq.com/plugins: key-auth-example, rate-limiting-base, rate-limiting-authenticated
spec:
  ports:
    - port: 1025
      name: tcp
      protocol: TCP
      targetPort: 1025
    - port: 1026
      name: udp
      protocol: TCP
      targetPort: 1026
    - port: 1027
      name: http
      protocol: TCP
      targetPort: 1027
  selector:
    app: echo
---
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: echo
  name: echo
spec:
  replicas: 1
  selector:
    matchLabels:
      app: echo
  strategy: {}
  template:
    metadata:
      labels:
        app: echo
    spec:
      containers:
        - image: kong/go-echo:latest
          name: echo
          ports:
            - containerPort: 1027
          env:
            - name: NODE_OWNER
              value: Michael
            - name: NODE_NAME
              valueFrom:
                fieldRef:
                  fieldPath: spec.nodeName
            - name: POD_NAME
              valueFrom:
                fieldRef:
                  fieldPath: metadata.name
            - name: POD_NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
            - name: POD_IP
              valueFrom:
                fieldRef:
                  fieldPath: status.podIP
          resources: {}
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: echo
  annotations:
    konghq.com/strip-path: "true"
spec:
  ingressClassName: kong
  rules:
    - http:
        paths:
          - path: /echo
            pathType: ImplementationSpecific
            backend:
              service:
                name: echo
                port:
                  number: 1027
---
apiVersion: v1
kind: Secret
metadata:
  name: alex-key-auth
  labels:
    konghq.com/credential: key-auth
stringData:
  key: hello_world
---
apiVersion: configuration.konghq.com/v1
kind: KongConsumer
metadata:
  name: alex
  annotations:
    kubernetes.io/ingress.class: kong
    # konghq.com/plugins: rate-limiting-authenticated
username: alex
credentials:
  - alex-key-auth
---
apiVersion: v1
kind: Secret
metadata:
  name: bob-key-auth
  labels:
    konghq.com/credential: key-auth
stringData:
  key: bob_password
---
apiVersion: configuration.konghq.com/v1
kind: KongConsumer
metadata:
  name: bob
  annotations:
    kubernetes.io/ingress.class: kong
username: bob
credentials:
  - bob-key-auth
---
apiVersion: configuration.konghq.com/v1
kind: KongPlugin
metadata:
  name: key-auth-example
plugin: key-auth
config:
  key_names:
    - apikey
---
apiVersion: configuration.konghq.com/v1
kind: KongPlugin
metadata:
  name: rate-limiting-base
plugin: rate-limiting
config:
  second: 1
  policy: local
---
apiVersion: configuration.konghq.com/v1
kind: KongPlugin
metadata:
  name: rate-limiting-authenticated
plugin: rate-limiting
config:
  second: 5
  policy: local
```

Reply:
@mheap I've rewritten the guide building on top of your example, slightly modifying it by using two `HTTPRoute`s and duplicating plugins on one of the routes. Please take a look: https://github.com/Kong/docs.konghq.com/pull/7463/commits/4ef76e251e6e8e6761fbab8d091792a57d90a1b8